Makefile

```diff
@@ -229,6 +232,12 @@ build/assets/%.o: assets/%.c
 	$(CC) -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o $@ $<
 	$(OBJCOPY) -O binary $@ $@.bin
 
+build/include/tables/dmadata_table.h: build/$(SPEC)
+	$(MKDMADATA) $< $@
+
+build/src/boot/z_std_dma.o: build/include/tables/dmadata_table.h
+build/src/dmadata/dmadata.o: build/include/tables/dmadata_table.h
```

Comment:
Very minor but this should probably be wrapped in a `ifeq ($(NON_MATCHING),1)` so it doesn't trigger rebuilds for matching builds

Reply:
Not sure I can do this, since `CC_CHECK` fails as it always builds with `NON_MATCHING`, so the generated file needs to present for it

Reply:
Yeah that's true. In that case maybe we should wrap the spec dmadata recipe as well in the if, and have a recipe for it in the else which just outputs a default file so CC_CHECK doesn't fail. Or perhaps we should have a C define specific to using spec dmadata instead of NON_MATCHING, which would be enabled if NON_MATCHING==1 in the makefile. 
I'm not sure what's the best option, but I feel like it's important to negate the impact for matching builds where the spec dmadata goes completely unused.

Reply:
Thinking about it more, this stuff might not matter to the build performance overall, especially once we start making relocs based on spec (which would trigger its own rebuilds based on changes to the spec).
So assuming mkdmadata is almost instant, we could probably leave it always running along with these rebuilds, and it would have the one advantage of actually running during normal builds so errors don't go silent.

Reply:
mkdmadata is essentially instant, I get on average 0.007s/0.008s when timing:
`time ./tools/mkdmadata build/spec build/dmadata_table_spec.h`
so it should be consistently low for all kinds of systems
```
real    0m0.007s
user    0m0.000s
sys     0m0.000s
```